File: WallPaper/src/m/k/android/sample/wallpaper/MainActivity.java
package m.k.android.sample.wallpaper;

import android.os.Bundle;
import android.app.Activity;
import android.app.WallpaperManager;
import android.content.Intent;
import android.view.View;

public class MainActivity extends Activity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void onClick(View view) {
    	switch(view.getId()) {
    		case R.id.btn1:
    			selectWallpaper();
    			break;
    	}
    }
    
    private void selectWallpaper() {
    	final Intent intent = new Intent();
    	intent.setAction(WallpaperManager.ACTION_LIVE_WALLPAPER_CHOOSER);
    	startActivity(intent);
    }
}

File: Sample/MultipleDeviceSupportSample/app/src/main/java/com/github/mag0716/multipledevicesupportsample/MainActivity.kt
package com.github.mag0716.multipledevicesupportsample

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import com.github.mag0716.multipledevicesupportsample.ui.license.LicenseActivity
import com.github.mag0716.multipledevicesupportsample.ui.settings.SettingsActivity

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        menuInflater.inflate(R.menu.menu, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.settings -> {
                startActivity(Intent(this, SettingsActivity::class.java))
            }
            R.id.license -> {
                startActivity(Intent(this, LicenseActivity::class.java))
            }
        }
        return false
    }
}